packages/core/src/lib/implementation/read-rc-file.integration.test.ts

```diff
@@ -0,0 +1,62 @@
+import { dirname, join } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { describe, expect } from 'vitest';
+import { readRcByPath } from './read-rc-file';
+
+describe('readRcByPath', () => {
+  const configDirPath = join(
+    fileURLToPath(dirname(import.meta.url)),
+    '..',
+    '..',
+    '..',
+    '..',
+    '..',
+    'testing-utils',
+    'src',
+    'lib',
+    'fixtures',
+    'configs',
+  );
+
+  it('should load the configuration', async () => {
```

Comment:
As a result of the comment above, this test should be replaced by the original trio of tests, one happy path per type.

Reply:
as the mocked API also throws it is valid imo